flexmeasures/data/models/time_series.py

```diff
@@ -756,7 +777,7 @@ def search(
                 else:
                     bdf = bdf.for_each_belief(get_median_belief)
 
-            if resolution is not None:
+            if resolution is not None and resolution != bdf.event_resolution:
```

Comment:
`resolution` can be `str` or `timedelta` which means that if it comes in `str`, it will always be different.

Reply:
I'm adding this as a note, but keeping the logic, with an eye on `str` type `bdf.event_resolution` in the future.